### DISCUSSION TITLE
Q: tsh for macOS 10.15 and teleport 12?

### DISCUSSION THREAD

**Discussion by furrnace** [gravitational/teleport#25374] [Q&A] [answered]
Is there a version of tsh that is compatible with macOS 10.15 and teleport 12?

**Comment by webvictim** [comment]
The regular `tsh` download on the website was built with Go 1.20 and should therefore run on MacOS 10.13+: https://cdn.teleport.dev/tsh-12.2.5.pkg

https://goteleport.com/download/#install-links

**Comment by furrnace** [comment]
I think that may be true in theory. In reality it doesn't work:

<img width="791" alt="Screenshot 2023-05-01 at 18 48 48" src="https://user-images.githubusercontent.com/110343153/235490932-31ddc556-8470-46bb-959a-e79d5f434ab6.png">


getting a version that works on macOS 10.15 would be highly appreciated.

**Reply by webvictim** [reply, marked as answer]
Ah - this is quite likely to be because the app is now signed/notarized to use entitlements for Touch ID functionality and xcode probably requires newer bindings. I forgot how old MacOS 10.15 is (late 2019)

You can try compiling from source if you like, but you won't be able to use Touch ID. You may need some other dependencies too. To try, install Go and make sure `go version` gives you output (i.e. `go` is on your path), then download the source, check out the tag and compile:

```console
git clone https://github.com/gravitational/teleport.git
cd teleport
git checkout v12.2.5
make build/tsh
```

**Reply by furrnace** [reply]
Thanks, I compiled it and I can run it now.
